pandas/_testing/asserters.py

```diff
@@ -610,6 +609,34 @@ def assert_interval_array_equal(
     assert_attr_equal("inclusive", left, right, obj=obj)
 
 
+def assert_interval_array_almost_equal(left, right, rtol=1e-5, atol=1e-8) -> None:
```

Comment:
Can't these tolerance arguments just be added to `assert_interval_array_equal`? Why is a separate function necessary?

Reply:
`assert_interval_array_equal` calls `assert_equal` which checks that the numbers being compared are exactly equal. Hence the decision to use assert_interval_array_almost_equal that takes tolerance and calls assert_almost_equal

Reply:
`assert_equal` takes `**kwargs` which are passed into the other `assert_*` functions that appear to accept `rtol` and `atol` so I imagine it should be possible to integrate this directly into `assert_interval_array_equal`

Reply:
`assert_equal` calls `assert_numpy_array_equal` which doesn't have `rtol` or `atol` parameters.
Within `assert_numpy_array_equal`, `if not array_equivalent` runs and calls `array_equivalent` (no **kwargs passed in)
This is where the error happens:
```
if is_float_dtype(left.dtype) or is_complex_dtype(left.dtype):
        if not (left.size and right.size):
            return True
        return ((left == right) | (isna(left) & isna(right))).all()
```
No tolerance is passed in, and it evaluates that left != right.

Reply:
Could `assert_interval_array_equal` dispatch to `assert_almost_equal` if tolerances are specified?

Reply:
So it would look something like:
```
def assert_interval_array_equal(
    left, right, exact="equiv", obj="IntervalArray", **kwargs
) -> None:
```
...
```
if kwargs['rtol'] or kwargs['atol']:
    assert_almost_equal(
        left._left, right._left, rtol=rtol, atol=atol, obj=f"{obj}.left"
    )
    assert_almost_equal(
        left._right, right._right, rtol=rtol, atol=atol, obj=f"{obj}.left"
    )
    assert_attr_equal("inclusive", left, right, obj="IntervalArray")
    return
kwargs = {}
    if left._left.dtype.kind in ["m", "M"]:
        # We have a DatetimeArray or TimedeltaArray
        kwargs["check_freq"] = False
```

Reply:
Yea similar to this.

Reply:
👍🏾
How do I address situations where only one of `rtol` or `atol` is passed in?

Reply:
You can just pass through the kwargs like `assert_almost_equal(left._left, right._left, **kwargs)`

Reply:
Got it.
Thanks.

Reply:
It looks like I may have to set a default `rtol` and `atol`.
`assert_frame_equal` takes in tolerance and calls `assert_series_equal` which also takes in tolerance, but tolerance isn't passed in when `assert_interval_array_equal` is called and I can't pass in `**kwargs` because it's not defined. 
If I pass in `rtol=rtol, atol=atol`, then the branching logic we want to implement in `assert_intterval_array_equal` will always evaluate to be true.
What should I do in this case?

Reply:
Yeah `rtol` and `atol` may need to be added explicitly in `assert_interval_array_equal` and other places

Reply:
So I should stick to using `assert_equal` and pass tolerance in to the end of the various function calls? Should I abandon using `assert_allmost_equal`?

Reply:
`assert_equal` would be ideal to minimize the code diff. In the end, we want to avoid creating a whole new function for testing intervals with tolerances.

Reply:
Thanks for all your comments @mroeschke ! I've made the edits in a new pull request [here](https://github.com/pandas-dev/pandas/pull/48295)

Reply:
Please just update this pull request instead of opening up a new pull request.

Reply:
Apologies 😬. I'll update this PR shortly

Reply:
On the previous new pull request I opened that got closed, Jeff mentioned that `array_equivalent` is meant for testing equality and not closeness, therefore I can't pass tolerance into it. What alternative approach should I take?

Reply:
No worries. The alternative is to use `assert_almost_equal` in place of `assert_equal` if possible.

Reply:
Got it! Thanks.